lib/git-fastclone.rb

```diff
@@ -343,18 +335,24 @@ def prefetch(submodule_file)
     # that this repo has been updated on this run of fastclone
     def store_updated_repo(url, mirror, repo_name, fail_hard)
       unless Dir.exist?(mirror)
-        Terrapin::CommandLine.new('git clone', '--mirror :url :mirror')
-                             .run(url: url.to_s, mirror: mirror.to_s)
+        fail_pipe_on_error(['git', 'clone', verbose ? '--verbose' : '--quiet', '--mirror', url.to_s, mirror.to_s], quiet: !verbose)
       end
 
-      Terrapin::CommandLine.new('cd', ':path; git remote update --prune').run(path: mirror)
-
+      Dir.chdir(mirror) {
+        output = fail_on_error(*['git', 'remote', verbose ? '--verbose' : nil, 'update', '--prune'].compact, quiet: !verbose)
+        puts "Output from `git remote update`:\n#{output}" if verbose
+      }
       reference_updated[repo_name] = true
-    rescue Terrapin::ExitStatusError => e
-      # To avoid corruption of the cache, if we failed to update or check out we remove
-      # the cache directory entirely. This may cause the current clone to fail, but if the
-      # underlying error from git is transient it will not affect future clones.
-      FileUtils.remove_entry_secure(mirror, force: true)
+    rescue RunnerExecutionRuntimeError => e
+      if retriable_error?(e.output)
+        print_formatted_error(e.output, mirror)
+        clear_cache(mirror, url)
+
+        if attempt_number < retries_allowed
+          attempt_number += 1
+          retry
```

Comment:
Might want to add some form of exponential backoff here to try and reduce thundering herd issues?

Reply:
discussed and this change is now out of scope